### PR TITLE
Feature match

### PR DESCRIPTION
**Pull request by jorol**
Added a method `match()` to PICA::Path, which supports options like `join`, `pluck`, `split` & `nested_arrays` and returns data like `marc_map()`. Added also a section 'Matching rules' to POD.

### CODE REVIEW

File: lib/PICA/Path.pm [hunk 1]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+        record => [
+            [ '005A', '', '0', '1234-5678' ],
+            [ '005A', '', '0', '1011-1213' ],
+            [   '009Q', '', 'u', 'http://example.org/', 'x', 'A', 'z', 'B', 'z',
+                'C'
+            ],
+            [ '021A', '', 'a', 'Title', 'd', 'Supplement' ],
+            [   '031N', '',     'j', '1600', 'k', '1700',
+                'j',    '1800', 'k', '1900', 'j', '2000'
+            ],
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Ugly line breaks in this example.

File: lib/PICA/Path.pm [hunk 2]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
 Stringifies the PICA path to normalized form. Subfields are separated with
 C<$>, unless called as C<stringify(1)> or the first subfield is C<$>.
 
+=head1 Matching rules
+
+    ...
```
Comment (nichtich) [inline lib/PICA/Path.pm]:
~~~
=head1 MATCHING RULES

Matching rules are illustrated with the following sample record:
~~~

And link to this section from INTRODUCTION above or move the whole section into INTRODUCTION

File: lib/PICA/Path.pm [hunk 3]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+    use PICA::Path;
+    use Test::More;
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Remove `Test::More`, no need to make this a test case

File: lib/PICA/Path.pm [hunk 4]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+
+Field C<021A> has only unique subfield codes:
+
+    $record = {
+        record => [
+            ...
+            [ '021A', '', 'a', 'Title', 'd', 'Supplement' ],
+            ...
+        ],
+    };
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Why duplicate the example from above?

File: lib/PICA/Path.pm [hunk 5]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+    
+    # create path
+    my $path = PICA::Path->new('021Aab');
+    
+    # match record
+    my $match = $path->match($record);
+    
+    # test result
+    is($match,'TitleSupplement');
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Append `  # $match = "TitleSupplement" to line 480 and remove the test call

File: lib/PICA/Path.pm [hunk 6]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+            ...
+            [ '021A', '', 'a', 'Title', 'd', 'Supplement' ],
+            ...
+        ],
+    };
+    
+    # get all subfields
+    $path = PICA::Path->new('021A');
+    $match = $path->match($record);
+    is($match, 'TitleSupplement');
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Remove test calls like above

File: lib/PICA/Path.pm [hunk 7]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+    # get subfields by code
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record);
+    is($match, '1234-56781011-1213');
+
+Option C<split> creates a list out of subfields:
+
+    # split subfields to list
+    $path = PICA::Path->new('005A0');
+    $match = $path->match($record, {split => 1});
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Why not allow slightly simpler call if optional argument is not a hasref?:

    $path->match($record, split => 1);

File: lib/PICA/Path.pm [hunk 8]
```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
...
+    
+    # get subfields by code
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record);
+    is($match, '1234-56781011-1213');
+
+Option C<split> creates a list out of subfields:
+
+    # split subfields to list
+    $path = PICA::Path->new('005A0');
```
Comment (nichtich) [inline lib/PICA/Path.pm]: I prefer to include the optional `$` to better make clear that `0` is a subfield code: `005A$0`.

File: lib/PICA/Path.pm [hunk 9]
```diff
@@ -241,16 +372,71 @@ supported.
 
 =back
 
-=head2 match_field( $field )
+=head2 match( $record, \%opts )
```
Comment (nichtich) [inline lib/PICA/Path.pm]: Allow non-ref hash with options:

    =head2 match( $record, \%opts | %opts )

**Comment by jorol** [comment]
Thanks for the comments. I will rework this. I also need to add an option 'apppend'/'force_array' option. I will close this PR for now.

**Pull request closed without merging**